api/integrations/common/serializers.py

```diff
@@ -0,0 +1,23 @@
+from rest_framework.serializers import ModelSerializer
+
+
+class _BaseIntegrationModelSerializer(ModelSerializer):
+    one_to_one_field_name = None
+
+    def create(self, validated_data):
+        if (related_obj := validated_data.get(self.one_to_one_field_name)) and (
```

Comment:
This if condition is doing a bit too much I think?  Maybe we can add a comment? Or move this if check to a different method? Named something like `get_existing_object`? What do you think?

Reply:
Good point - I have done a combination of the 2 things you suggested. Let me know what you think.

Reply:
Yeah, it looks perfect